sdk/confidentialledger/Azure.Security.ConfidentialLedger/README.md

```diff
@@ -199,31 +239,104 @@ ledgerClient.PostLedgerEntry(
 
 firstPostResponse.Headers.TryGetValue(ConfidentialLedgerConstants.Headers.TransactionId, out string transactionId);
 
+// Wait for the entry to be committed
+status = "Pending";
+while (status == "Pending")
+{
+    statusResponse = ledgerClient.GetTransactionStatus(transactionId);
+    status = JsonDocument.Parse(statusResponse.Content)
+        .RootElement
+        .GetProperty("state")
+        .GetString();
+}
+
 // The ledger entry written at the transactionId in firstResponse is retrieved from the default sub-ledger.
 Response getResponse = ledgerClient.GetLedgerEntry(transactionId);
+
+// Try until the entry is available.
+loaded = false;
+element = default;
+contents = null;
+while (!loaded)
+{
+    loaded = JsonDocument.Parse(getResponse.Content)
+        .RootElement
+        .TryGetProperty("entry", out element);
+    if (loaded)
+    {
+        contents = element.GetProperty("contents").GetString();
+    }
+    else
+    {
+        getResponse = ledgerClient.GetLedgerEntry(transactionId, subLedgerId);
+    }
+}
+
 string firstEntryContents = JsonDocument.Parse(getResponse.Content)
     .RootElement
+    .GetProperty("entry")
     .GetProperty("contents")
     .GetString();
 
 Console.WriteLine(firstEntryContents); // "Hello world 0"
 
 // This will return the latest entry available in the default sub-ledger.
 getResponse = ledgerClient.GetCurrentLedgerEntry();
-string latestDefaultSubLedger = JsonDocument.Parse(getResponse.Content)
-    .RootElement
-    .GetProperty("contents")
-    .GetString();
+
+// Try until the entry is available.
+loaded = false;
+element = default;
+string latestDefaultSubLedger = null;
+while (!loaded)
+{
+    loaded = JsonDocument.Parse(getResponse.Content)
+        .RootElement
+        .TryGetProperty("contents", out element);
+    if (loaded)
+    {
+        latestDefaultSubLedger = element.GetString();
+    }
+    else
+    {
+        getResponse = ledgerClient.GetCurrentLedgerEntry();
+    }
+}
 
 Console.WriteLine($"The latest ledger entry from the default sub-ledger is {latestDefaultSubLedger}"); //"Hello world 1"
 
 // The ledger entry written at subLedgerTransactionId is retrieved from the sub-ledger 'sub-ledger'.
 subLedgerPostResponse.Headers.TryGetValue(ConfidentialLedgerConstants.TransactionIdHeaderName, out string subLedgerTransactionId);
+
+// Wait for the entry to be committed
+status = "Pending";
+while (status == "Pending")
+{
+    statusResponse = ledgerClient.GetTransactionStatus(subLedgerTransactionId);
+    status = JsonDocument.Parse(statusResponse.Content)
+        .RootElement
+        .GetProperty("state")
+        .GetString();
+}
+
 getResponse = ledgerClient.GetLedgerEntry(subLedgerTransactionId, "my sub-ledger");
-string subLedgerEntry = JsonDocument.Parse(getResponse.Content)
-    .RootElement
-    .GetProperty("contents")
-    .GetString();
+// Try until the entry is available.
```

Comment:
Should this appear in the snippet?

Reply:
Yes, I wanted to demonstrate that the results are not necessarily immediately available. I know it's ugly, but it's a short term solution.